### PR TITLE
Bundler: Add tests for bundler 2 resolution

### PR DESCRIPTION
**Pull request by feelepxyz**
Add UpdateChecker test cases for a gem that depends on a particular
version of bundler.

### CODE REVIEW

**Comment by jurre** [comment]
Is this a duplicate of https://github.com/dependabot/dependabot-core/pull/3418?

**Review by jurre — APPROVED** [review]

**Comment by feelepxyz** [comment]
@jurre oh sorry totally missed your PR! 🤦

File: bundler/spec/dependabot/bundler/update_checker_spec.rb [hunk 1]
```diff
@@ -1412,6 +1412,23 @@
...
+
+    context "with a gem that depends on bundler" do
+      let(:dependency_files) { bundler_project_dependency_files("guard_bundler") }
+      let(:requirements) do
+        [{ file: "Gemfile", requirement: "~> 2.2.1, <= 3.0.0", groups: [], source: nil }]
+      end
+      let(:dependency_name) { "guard-bundler" }
+      let(:current_version) { "2.2.1" }
+
+      context "using bundler v1", :buldler_v1_only do
```
Comment (feelepxyz) [inline bundler/spec/dependabot/bundler/update_checker_spec.rb:1424]: ops misspelt the key `buldler_v1_only` 😀

**Comment by jurre** [comment]
> @jurre oh sorry totally missed your PR! 🤦

No worries 😄 let's go with this one though 👍

**Pull request merged**